test/StringProcessing/SILGen/regex_literal_silgen.swift

```diff
@@ -11,4 +11,4 @@ var s = '/abc/'
 // CHECK: [[VERSION_INT:%[0-9]+]] = apply [[INT_INIT]]([[VERSION_LITERAL]]
 
 // CHECK: [[REGEX_INIT:%[0-9]+]] = function_ref @$s17_StringProcessing5RegexV06_regexA07versionACyxGSS_SitcfC : $@convention(method) <τ_0_0> (@owned String, Int, @thin Regex<τ_0_0>.Type) -> @out Regex<τ_0_0>
-// CHECK: apply [[REGEX_INIT]]<DynamicCaptures>({{%.+}}, [[REGEX_STR]], [[VERSION_INT]], {{%.+}}) : $@convention(method) <τ_0_0> (@owned String, Int, @thin Regex<τ_0_0>.Type) -> @out Regex<τ_0_0>
+// CHECK: apply [[REGEX_INIT]]<()>({{%.+}}, [[REGEX_STR]], [[VERSION_INT]], {{%.+}}) : $@convention(method) <τ_0_0> (@owned String, Int, @thin Regex<τ_0_0>.Type) -> @out Regex<τ_0_0>
```

Comment:
Would these lit tests start failing for someone who pulls swift-main but not the package?

Reply:
I don't believe we should guarantee "no test failures" esp when we change capture type inference behavior.  What we do try to guarantee is "no build failures" for a mismatched checkout.

For this specific SIL test, a wildcard match for the capture type could be reasonable.